src/main/java/gyro/google/compute/ComputeResource.java

```diff
@@ -16,67 +16,50 @@
 
 package gyro.google.compute;
 
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
-import com.google.api.client.googleapis.json.GoogleJsonResponseException;
 import com.google.api.services.compute.Compute;
 import com.google.api.services.compute.model.Operation;
 import gyro.core.GyroException;
+import gyro.core.Wait;
 import gyro.google.GoogleResource;
 
 public abstract class ComputeResource extends GoogleResource {
 
-    public void waitForCompletion(Compute compute, Operation operation) throws Exception {
-        waitForCompletion(compute, operation, 60000);
-    }
-
-    public void waitForCompletion(Compute compute, Operation operation, long timeout) throws Exception {
-        long start = System.currentTimeMillis();
-        final long pollInterval = 1000;
-
-        String zone = operation.getZone();
-        if (zone != null) {
-            String[] bits = zone.split("/");
-            zone = bits[bits.length - 1];
-        }
+    public void waitForCompletion(Compute compute, Operation operation) {
```

Comment:
This method should call another method that takes in the time and time unit.
That method should implement the actual wait logic.

This way we will have option to use a custom time for the wait if needed be.